### PR TITLE
Add upgrade guide paragraph about XML failing parser configuration

### PR DESCRIPTION
**Pull request by eskatos**
And enhance `XMLFactories` error messages.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 1]
```diff
@@ -44,6 +44,13 @@ The previous step will help you identify potential problems by issuing deprecati
 
 The embedded Kotlin has been updated to link:https://github.com/JetBrains/kotlin/releases/tag/v1.9.10[Kotlin 1.9.10].
 
+==== XML parsing now requires recent parsers
+
+Gradle 8.4 now configures XML parsers with security features enabled.
+If your build logic has dependencies on old XML parsers that don't support secure parsing your build may now fail.
+You need to make sure your build logic doesn't depend on too old XML parsers.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:51]: ❓ That second sentence does not add much.
What about:
"If you encounter a failure, check and update or remove any dependency on legacy XML parsers."

**Review by ljacomet — CHANGES_REQUESTED** [review]
One possible improvement to the note, see below

I also fixed the milestone.

File: subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc (hunk 1 above)
Comment (eskatos) [reply subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc]: That's better. Done

**Review by ljacomet — APPROVED** [review]

**Comment by ljacomet** [comment]
@bot-gradle test and merge

**Review by lkasso — APPROVED** [review]

File: subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 2]
```diff
@@ -44,6 +44,13 @@ The previous step will help you identify potential problems by issuing deprecati
 
 The embedded Kotlin has been updated to link:https://github.com/JetBrains/kotlin/releases/tag/v1.9.10[Kotlin 1.9.10].
 
+==== XML parsing now requires recent parsers
+
+Gradle 8.4 now configures XML parsers with security features enabled.
+If your build logic has dependencies on old XML parsers that don't support secure parsing your build may now fail.
```
[the remark this answers is not in the document]
Comment (eskatos) [reply subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:50]: Thanks. Fixed in https://github.com/gradle/gradle/pull/26535

**Comment by ben-manes** [comment]
Please consider adding this workaround hint ([#26672](https://github.com/gradle/gradle/issues/26672)) for confused users when they encounter the error.

**Pull request merged**